addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/ws/IhcClient.java

```diff
@@ -0,0 +1,558 @@
+     */
+    public synchronized WSSystemInfo getSystemInfo() throws IhcExecption {
+        return configurationService.getSystemInfo();
+    }
+
+    /**
+     * Query detected RF devices from the controller.
+     *
+     * @return List of RF devices.
+     * @throws IhcExecption
+     */
+    public synchronized List<WSRFDevice> getDetectedRFDevices() throws IhcExecption {
+        return airlinkManagementService.getDetectedDeviceList();
+    }
+
+    /**
+     * Query controller current state.
+     *
+     * @return controller's current state.
+     */
+    public WSControllerState getControllerState() throws IhcExecption {
+        return controllerService.getControllerState();
+    }
+
+    /**
+     * Query project number of segments.
+     *
+     * @return number of segments.
+     */
+    public int getProjectNumberOfSegments() throws IhcExecption {
+        return controllerService.getProjectNumberOfSegments();
+    }
+
+    /**
+     * Query project segmentation size.
+     *
+     * @return segmentation size in bytes.
+     */
+    public int getProjectSegmentationSize() throws IhcExecption {
+        return controllerService.getProjectSegmentationSize();
+    }
+
+    /**
+     * Query project segments data.
+     *
+     * @return segments data.
+     */
+    public WSFile getProjectSegment(int index, int major, int minor) throws IhcExecption {
+        return controllerService.getProjectSegment(index, major, minor);
+    }
+
+    /**
+     * Fetch project file from controller.
+     *
+     * @return project file.
+     */
+    public byte[] loadProjectFileFromControllerAsByteArray() throws IhcExecption {
+        try {
+
+            WSProjectInfo projectInfo = getProjectInfo();
+            int numberOfSegments = getProjectNumberOfSegments();
+            int segmentationSize = getProjectSegmentationSize();
+
+            logger.debug("Number of segments: {}", numberOfSegments);
+            logger.debug("Segmentation size: {}", segmentationSize);
+
+            ByteArrayOutputStream byteStream = new ByteArrayOutputStream();
+
+            for (int i = 0; i < numberOfSegments; i++) {
+                logger.debug("Downloading segment {}", i);
+
+                WSFile data = getProjectSegment(i, projectInfo.getProjectMajorRevision(),
+                        projectInfo.getProjectMinorRevision());
+                byteStream.write(data.getData());
+            }
+
+            logger.debug("File size before base64 encoding: {} bytes", byteStream.size());
+            byte[] decodedBytes = javax.xml.bind.DatatypeConverter.parseBase64Binary(byteStream.toString());
+            logger.debug("File size after base64 encoding: {} bytes", decodedBytes.length);
+            GZIPInputStream gzis = new GZIPInputStream(new ByteArrayInputStream(decodedBytes));
+            InputStreamReader in = new InputStreamReader(gzis, "ISO-8859-1");
+            return IOUtils.toByteArray(in, "ISO-8859-1");
+        } catch (Exception e) {
```

Comment:
Better catch specific exceptions (unless some method throws Exception). Using generic Exception also catches InterruptException and you need to reset the interrupt state then,